vmm/src/device_manager.rs

```diff
@@ -2762,6 +2768,21 @@ impl DeviceManager {
             });
         }
 
+        // Register DMA mapping in IOMMU.
+        // Do not register virtio-mem regions, as they are handled directly by
+        // virtio-mem device itself.
+        for (_, zone) in self.memory_manager.lock().unwrap().memory_zones().iter() {
+            for region in zone.regions() {
```

Comment:
How do you know it's not a virtio-mem region?

Reply:
Because of the way a MemoryZone is defined:
```rust
pub struct MemoryZone {
    regions: Vec<Arc<GuestRegionMmap>>,
    virtio_mem_zone: Option<VirtioMemZone>,
}

impl MemoryZone {
    pub fn regions(&self) -> &Vec<Arc<GuestRegionMmap>> {
        &self.regions
    }
    pub fn virtio_mem_zone(&self) -> &Option<VirtioMemZone> {
        &self.virtio_mem_zone
    }
}
```
You get access to the virtio-mem region from the `VirtioMemZone`.